### PR TITLE
Do not accept `--parachain-id` and `--chain` together

### PR DESCRIPTION
**Pull request by bkchr**

### CODE REVIEW

**Comment by JoshOrndorff** [comment]
This is because the parachain id is supposed to come from the spec, right?

**Comment by bkchr** [comment]
`--parachain-id` is more like a hack anyway. I don't see why any real parachain should support different parachain ids from the CLI level.

The genesis state is just the genesis header and if you pass `--chain`, we will open the spec to get the header.

The parachain id is only required to generate the spec, when we don't use any named `--chain`, because the parachain id is passed to the genesis config.

**Comment by JoshOrndorff** [comment]
> The parachain id is only required to generate the spec

Should it be associated with the `build-spec` subcommand then?

**Comment by bkchr** [comment]
> > The parachain id is only required to generate the spec
> 
> Should it be associated with the `build-spec` subcommand then?

Depends on your chain.

In the example chain here, yes it could be added, but I don't see the need.

File: rococo-parachains/src/cli.rs [hunk 1]
```diff
@@ -60,15 +60,15 @@ pub struct ExportGenesisStateCommand {
 	pub output: Option<PathBuf>,
 
 	/// Id of the parachain this state is for.
-	#[structopt(long, default_value = "100")]
+	#[structopt(long, default_value = "100", conflicts_with = "chain")]
```
Comment (jasl) [inline rococo-parachains/src/cli.rs:63]: `parachain_id` has default value which means you can't pass `chain` arg because `parachain_id` always has value

Comment (bkchr) [reply rococo-parachains/src/cli.rs:63]: Does it complain at runtime?

Comment (jasl) [reply rococo-parachains/src/cli.rs:63]: I don't know what your means, but you can simply verify what I said

```
jasl@localhost:~/Workspaces/BlockChain/cumulus on master$ target/release/rococo-collator export-genesis-state --chain tick
error: The argument '--chain <chain>' cannot be used with '--parachain-id <parachain-id>'

USAGE:
    rococo-collator export-genesis-state --chain <chain> --parachain-id <parachain-id>

For more information try --help
```

Comment (bkchr) [reply rococo-parachains/src/cli.rs:63]: Yeah that is at runtime. Hmmm. That is bad.

**Pull request merged**